velox/functions/prestosql/window/Rank.cpp

```diff
@@ -75,10 +127,47 @@ void registerRank(const std::string& name) {
       [name](
           const std::vector<TypePtr>& /*argTypes*/,
           const std::vector<column_index_t>& /*argIndices*/,
-          const TypePtr& /*resultType*/,
+          const TypePtr& resultType,
```

Comment:
There is another way to write this code that would look good. 
We would write a template function  (with the enum type) called registerRank. It takes the resultType also as a parameter. This registration function creates the signature and calls registerWindowFunction (using the enum template).

The registerRank, DenseRank and PercentRank just calls the registerRank function with the correct resultType and template enum value. 

The new code would avoid the duplication of the signature and registration calls.

Please can you try that.

Reply:
Thanks for the suggestion, this is so much more concise and have modified it accordingly!